python/vineyard/contrib/ml/tensorflow.py

```diff
@@ -145,6 +156,34 @@ def tf_table_resolver(obj, resolver):
     return tf.data.Dataset.from_tensor_slices((dict(arrow), labels))
 
 
+def tf_global_tensor_resolver(obj, **kw):
+    meta = obj.meta
+    endpoint = kw.get('endpoint')
+    client = vineyard.connect(endpoint)
+    num = from_json(meta['num'])
+    partition_index = from_json(meta['partition_index_'])
+    data = []
+    for i in range(partition_index):
+        loc = meta[f'partition_{i}']
+        if loc.islocal:
+            data.append(client.get(loc.id))
+    return data
+
+
+def tf_global_dataframe_resolver(obj, **kw):
+    meta = obj.meta
+    endpoint = kw.get('endpoint')
+    client = vineyard.connect(endpoint)
+    num = from_json(meta['num'])
+    partition_index = from_json(meta['partition_index_'])
+    data = []
+    for i in range(partition_index):
+        loc = meta[f'partition_{i}']
+        if loc.islocal:
+            data.append(client.get(loc.id))
+    return data
```

Comment:
Could you check the from_generator method (https://www.tensorflow.org/api_docs/python/tf/data/Dataset#from_generator) and return a single dataset instead of a list of datasets. So as the case in global tensor resolver.

Reply:
Used tf.concatenate , it will also return a single dataset.